crates/vault-registry/src/lib.rs

```diff
@@ -417,6 +417,21 @@ pub mod pallet {
             Self::_set_liquidation_collateral_threshold(currency_pair, threshold);
             Ok(())
         }
+
+        /// Changes the collateral liquidation threshold for a currency (only executable by the Root account)
+        ///
+        /// # Arguments
+        /// * `version` - the semver version, where the zero-index element is the major version
+        /// * `checksum` - the SHA256 checksum of the client binary
+        #[pallet::weight(<T as Config>::WeightInfo::set_liquidation_collateral_threshold())]
+        #[transactional]
+        pub fn set_client_release(origin: OriginFor<T>, version: [u32; 3], checksum: [u8; 32]) -> DispatchResult {
```

Comment:
When is governance supposed to make this call? IMO it would make the most sense on the actual runtime upgrade call but this is in the best case 4 hours before the upgrade goes live and in the worst case two weeks before the upgrade is enacted.
This should be called with the enactment of a runtime upgrade if the client version is supposed to change otherwise, this is subject to a separate governance proposal that someone needs to do and that might not be executed.

Reply:
I've noted some thoughts [here](https://github.com/interlay/interbtc/issues/672), let's move the discussion there. Some notes specific to this PR:
- I think we should directly set the client URL directly to avoid ambiguity
- `set_client_release` should be called for a *future* upgrade to allow "Vaultvisor" some time to download the new release

Reply:
Had a call with @savudani8 and @sander2, the plan is as follows:
- add `set_pending_client_release` or so to notify scheduled upgrade (batched alongside `authorize_upgrade`)
- add `set_current_client_release` or so to override current version - in case of bug fix, etc.
- these require two separate storage items `PendingClientRelease` and `CurrentClientRelease`